.gitlab/hpsf-gitlab-ci.yml

```diff
@@ -39,3 +39,30 @@ INTEL-DATA-CENTER-MAX-1100:
       - build/result_pvc1100.xml
     reports:
       junit: build/result_pvc1100.xml
+
+AMD-MI300A:
+  stage: test
+  tags: [uo-gpu, odyssey, amd-mi300]
+  image: rocm/dev-ubuntu-24.04:6.2.4-complete
+  script:
+    - sudo apt-get update && sudo apt-get install -y cmake
+    - export GTEST_FILTER=-hip.unified_memory_zero_memset
```

Comment:
I don't know. That's the first time for me to run this unit test. The backtrace looks like
```
#0  Kokkos::Impl::ParallelFor<Kokkos::Impl::zero_with_hip_kernel(Kokkos::HIP const&, void*, unsigned long)::{lambda(unsigned long)#1}, Kokkos::RangePolicy<Kokkos::HIP>, Kokkos::HIP>::operator()() const (
    this=<optimized out>) at /home/users/darndt/kokkos/core/src/HIP/Kokkos_HIP_ParallelFor_Range.hpp:67
#1  Kokkos::Impl::hip_parallel_launch_local_memory<Kokkos::Impl::ParallelFor<Kokkos::Impl::zero_with_hip_kernel(Kokkos::HIP const&, void*, unsigned long)::{lambda(unsigned long)#1}, Kokkos::RangePolicy<Kokkos::HIP>, Kokkos::HIP>, 1024u, 1u>(Kokkos::Impl::ParallelFor<Kokkos::Impl::zero_with_hip_kernel(Kokkos::HIP const&, void*, unsigned long)::{lambda(unsigned long)#1}, Kokkos::RangePolicy<Kokkos::HIP>, Kokkos::HIP>) (
    driver=...) at /home/users/darndt/kokkos/core/src/HIP/Kokkos_HIP_KernelLaunch.hpp:84
```
The point of this test was that `ZeroMemset` would fail and running a kernel was the workaround but that doesn't seem to work here either.

Reply:
`hipMemsetAsync` gives
```
C++ exception with description "hipMemsetAsync(dst, 0, cnt, exec_space.hip_stream()) error( hipErrorInvalidValue): invalid argument /home/users/darndt/kokkos/core/src/HIP/Kokkos_HIP_ZeroMemset.hpp:39" thrown in the test body.
```

Reply:
>  Is that a known issue?

No it should work. Before disabling the test we need someone (@ldh4 ?) to check that we haven't broken the test again.

Reply:
@cwpearson Can you take another look?

Reply:
I still think that this issue shouldn't block this pull request.

Reply:
Agreed